src/core/unit.c

```diff
@@ -1896,8 +1902,16 @@ int unit_start(Unit *u, ActivationDetails *details) {
                 return -EAGAIN;
 
         /* Units that aren't loaded cannot be started */
-        if (u->load_state != UNIT_LOADED)
-                return -EINVAL;
+        if (u->load_state != UNIT_LOADED) {
+                if (MANAGER_IS_TEST_RUN(u->manager)) {
+                        assert(u->load_state == UNIT_STUB);
+                        u->load_state = UNIT_LOADED;
```

Comment:
This doesn't seem right? There are also "fully-charged" test runs that go through usual unit load routines, e.g. analyze?

Reply:
Yeah, I concerned that. Also, e.g. test-execute does load units. But in that case, the state should be already in UNIT_LOADED, so I concluded that this is not a big matter.
If you do not like the change, I will drop the change, and call unit_patch_contexts() everywhere we need.

Reply:
OK. Dropped the change, and unit_patch_contexts() is called in tests where necessary.